modules/puppet_connect/plans/test_input_data.pp

```diff
@@ -59,6 +60,16 @@
         fail_plan("Inventory contains target ${target} with unsupported transport, must be ssh or winrm")
       }
     }
+
+    $plugin = $target.plugin_hooks["puppet_library"]
+    $user_configured_plugin = $plugin != { "plugin"=> "puppet_agent", "stop_service"=> true }
```

Comment:
A comment about how bolt-server uses a different default here would be helpful.

Reply:
Would it be worth changing the Bolt default for puppet-library to w/e bolt-server uses? So changing https://github.com/puppetlabs/bolt/blob/main/lib/bolt/plugin.rb#L145 to https://github.com/puppetlabs/bolt/blob/main/lib/bolt_server/transport_app.rb#L696. That would eliminate this check, and also the other case of what to do when the user-specified plugin _is_ the current Bolt default.

Reply:
The issue is that what bolt-server is doing is technically less correct. The proper thing to do is use `plugin => puppet_agent` which will then reference [`bolt-plugin.json`](https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/main/bolt_plugin.json) to determine which task plugin corresponds to the `puppet_library` hook. For bolt-server, we just hardcode that task to `puppet_agent::install` and 🤞 it doesn't change.

Reply:
Right, so I wonder if we should now hard-code that default to `puppet_agent::install` ? Sounds like no.